business/istio_status.go

```diff
@@ -0,0 +1,118 @@
+package business
+
+import (
+	"github.com/kiali/kiali/models"
+	apps_v1 "k8s.io/api/apps/v1"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+)
+
+// SvcService deals with fetching istio/kubernetes services related content and convert to kiali model
+type IstioStatusService struct {
+	k8s kubernetes.IstioClientInterface
+}
+
+type ComponentStatus struct {
+	// The app label value of the Istio component
+	//
+	// example: istiod
+	// required: true
+	Name string `json:"name"`
+
+	// The status of a Istio component
+	//
+	// example:  Not Found
+	// required: true
+	Status string `json:"status"`
+
+	// When true, the component is part of istio core. Otherwise, it is an addon
+	//
+	// example:  true
+	// required: true
+	IsCore bool `json:"is_core"`
+}
+
+type IstioComponentStatus []ComponentStatus
+
+const (
+	Healthy   string = "Healthy"
+	Unhealthy string = "Unhealthy"
+	NotFound  string = "NotFound"
+)
+
+// List of workloads part of a Istio deployment and if whether it is mandatory or not
+var components = map[string]bool{
+	// Core components, mandatory
+	"istio-egressgateway":  true,
+	"istio-ingressgateway": true,
+	"istiod":               true,
```

Comment:
Perhaps that will be for a next PR to split the work, but I think we'll want compatibility with older Istio components. Not only for istio < 1.5 but also because Istio can still run without istiod, with old mixer & co (as I just learnt here https://github.com/kiali/kiali/issues/2669#issuecomment-616596458 )

Reply:
Good catch. wdyt of the following behavior?

```
istiod  | pilot  | outcome
-------------------------------------
Not Found   | Not Found  | Error message: Pilot not found
Not Found   | Found      | Status for non-monolithic istio control plane stack (mixer and co)
Found       | Not Found  | Status for monolithic istio control plane stack (istiod)
Found       | Found      | Error message: Multiple Pilot installed
```

I think this will cover both scenarios. Even weird installations (two pilots) (because they are weird, right?)

Reply:
Looks good! I agree that having both istiod and pilot is probably not supposed to happen and can be reported as a problem (until proven otherwise)

Reply:
Thinking out loud. Does it make sense to show the 'Not found' component status?

Istio installation can be really diverse: ones using tracing, others don't; ones using grafana; others don't; ones egress, others don't and so on[1]. If we show the not found, the icon on the masthead might lose some meaning bc it will be always present and warning.

Therefore, what we could do is just show the status of the istio components (monolith and non-monolith ones) because we consider that users aren't deleting Istio components carelessly. 

Side effect, implementation will be easier: check status of Istio components that are deployed in the control plane namespace. If they are not present, don't show any status. Include both monolith and non-monolith components in the list.

Otherwise, this PR will be very demo profile oriented. IMO.

[1] Minimal or remote profiles are quite skinny: https://istio.io/docs/setup/additional-setup/config-profiles/

Reply:
Just a comment.
Let's try to differentiate errors/warnings.
Error: some situation we know that is cause a problem.
Warning: something that may show something unusual i.e. tracing not enabled. We know Istio may work, but probably Kiali will miss some feature.

Perhaps we can't cover all combinations of situations on a single PR.

This work was defined to offer a first feature that can be extended with more use cases.

But no need to cover all of them, it can go in following iterations if some of them is not trivial.

Reply:
About third party not found, I think we have to match that against Kiali config. If Jaeger is disabled in Kiali, we won't show the not found issue, but if it's enabled we should show it. Same for Grafana.

Reply:
Actually for Jaeger and Grafana, I don't know, if you choose to remove them from the scope of this task, I think it's fine as well because we have other obvious signals / messages that comes up when they're missing. So, it's not a big value added to have them here as well.

For other components, it's different. Having them missing can be a real problem that goes unnoticed in Kiali, without those checks.